File: src/main/java/kr/co/bookand/backend/bookstore/ReportBookStore.java
package kr.co.bookand.backend.bookstore;

import kr.co.bookand.backend.account.Account;
import kr.co.bookand.backend.common.BaseTimeEntity;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class ReportBookStore extends BaseTimeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    private String content;
    private String answer;

    @ManyToOne(fetch = FetchType.LAZY)
    private Account account;
}

File: src/main/kotlin/kr/co/bookand/backend/bookstore/model/ReportBookstore.kt
package kr.co.bookand.backend.bookstore.model

import kr.co.bookand.backend.account.model.Account
import kr.co.bookand.backend.bookstore.dto.AnswerReportRequest
import kr.co.bookand.backend.common.model.BaseEntity
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter
import javax.persistence.*

@Entity
class ReportBookstore(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "report_bookstore_id")
    var id: Long = 0,
    var name: String,
    val address: String,
    var isAnswered: Boolean,
    var answerTitle: String,
    var answerContent: String,
    var answeredAt: String,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "account_id")
    var account: Account? = null

) : BaseEntity() {
    fun updateAnswer(answerReportRequest: AnswerReportRequest) {
        isAnswered = true
        answerTitle = answerReportRequest.answerTitle
        answerContent = answerReportRequest.answerContent
        answeredAt = LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy/MM/dd HH:mm:ss"))
    }
}